### PR TITLE
[docs] Add a reference for Expo specific properties for `package.json`

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Fix ENG-14209 ENG-12667

# How

- Add a reference page under configuration files for **package.json**
- This page includes `install.exclude` info and a link `autolinking` reference page
- In Use libraries guide, add "Excluding a third-party library from version checks" section that links to package.json reference
- In Expo CLI reference, update "Configuring dependency validation" to link to package.json reference

# Test Plan

See preview: /versions/latest/config/package-json/

## Preview

**package.json reference**

CleanShot 2024-12-02 at 22 39 18

**Use libraries** 

CleanShot 2024-12-02 at 22 38 31

**Expo CLI reference**

CleanShot 2024-12-02 at 22 38 53

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/more/expo-cli.mdx [hunk 1]
```diff
@@ -408,23 +408,7 @@ The command `npx expo install expo-camera` and `npx expo install expo-camera --f
...
-```json package.json
-{
-  "expo": {
-    "install": {
-      "exclude": ["expo-updates", "expo-splash-screen"]
-    }
-  }
-}
-```
+There may be circumstances where you want to use a version of a package is different from the version recommended by `npx expo install`. In this case, you can exclude specific packages from version checking by using the [`expo.install.exclude`](/versions/latest/config/package-json/#exclude) property in your project's **package.json**.
```
Comment (keith-kurak) [inline docs/pages/more/expo-cli.mdx:411]:
```suggestion
There may be circumstances where you want to use a version of a package that is different from the version recommended by `npx expo install`. In this case, you can exclude specific packages from version checking by using the [`expo.install.exclude`](/versions/latest/config/package-json/#exclude) property in your project's **package.json**.
```

File: docs/pages/bare/installing-expo-modules.mdx [hunk 2]
```diff
@@ -168,7 +168,7 @@ To exclude any of these modules, refer to the following guide on [excluding modu
 
 ### Excluding specific modules from autolinking
 
-If you need to exclude Expo modules you are not using, which were installed by other dependencies, you can use the `expo.autolinking` field in **package.json**:
+If you need to exclude Expo modules you are not using, but were installed by other dependencies, you can use the [`expo.autolinking.exclude`](/modules/autolinking/#exclude) property in **package.json**:
```
Comment (keith-kurak) [inline docs/pages/bare/installing-expo-modules.mdx:171]:
```suggestion
If you need to exclude native code from Expo modules you are not using, but were installed by other dependencies, you can use the [`expo.autolinking.exclude`](/modules/autolinking/#exclude) property in **package.json**:
```
Trying to add a little precision on what it means to suppress autolinking

File: docs/pages/versions/unversioned/config/package-json.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+---
+title: package.json
+description: A reference for Expo specific properties that can be used in the package.json file.
```
Comment (keith-kurak) [inline docs/pages/versions/unversioned/config/package-json.mdx:3]:
```suggestion
description: A reference for Expo-specific properties that can be used in the package.json file.
```
I think the hyphen generally applies when another word modifies a noun to function as an adjective, but it sounds like maybe it's not 100% hard-and-fast? https://www.grammarly.com/blog/punctuation-capitalization/hyphen/

File: docs/pages/versions/unversioned/config/package-json.mdx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+isNew: true
+---
+
+**package.json** is a JSON file that contains the metadata for a JavaScript project. This is a reference to Expo specific properties that can be used in the **package.json** file.
```
Comment (keith-kurak) [inline docs/pages/versions/unversioned/config/package-json.mdx:7]:
```suggestion
**package.json** is a JSON file that contains the metadata for a JavaScript project. This is a reference to Expo-specific properties that can be used in the **package.json** file.
```

File: docs/pages/versions/unversioned/config/package-json.mdx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+## `install`
+
+### `exclude`
+
+The following commands perform a version check for the libraries installed in a project and give a warning when a library's version is different from the version recommended by Expo:
+
+- `npx expo start` and `npx expo-doctor`
+- `npx expo install` (when installing a new version of that library or using `--check` or `--fix` options)
+
+By specifying the library under `install.exclude` array in the **package.json** file, you can exclude it from the version checks:
```
Comment (keith-kurak) [inline docs/pages/versions/unversioned/config/package-json.mdx:18]:
```suggestion
By specifying the library under the `install.exclude` array in the **package.json** file, you can exclude it from the version checks:
```

File: docs/pages/versions/v52.0.0/config/package-json.mdx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+---
+title: package.json
+description: A reference for Expo specific properties that can be used in the package.json file.
```
Comment (keith-kurak) [inline docs/pages/versions/v52.0.0/config/package-json.mdx:3]:
```suggestion
description: A reference for Expo-specific properties that can be used in the package.json file.
```

File: docs/pages/versions/v52.0.0/config/package-json.mdx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+isNew: true
+---
+
+**package.json** is a JSON file that contains the metadata for a JavaScript project. This is a reference to Expo specific properties that can be used in the **package.json** file.
```
Comment (keith-kurak) [inline docs/pages/versions/v52.0.0/config/package-json.mdx:7]:
```suggestion
**package.json** is a JSON file that contains the metadata for a JavaScript project. This is a reference to Expo-specific properties that can be used in the **package.json** file.
```

File: docs/pages/versions/v52.0.0/config/package-json.mdx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+## `install`
+
+### `exclude`
+
+The following commands perform a version check for the libraries installed in a project and give a warning when a library's version is different from the version recommended by Expo:
+
+- `npx expo start` and `npx expo-doctor`
+- `npx expo install` (when installing a new version of that library or using `--check` or `--fix` options)
+
+By specifying the library under `install.exclude` array in the **package.json** file, you can exclude it from the version checks:
```
Comment (keith-kurak) [inline docs/pages/versions/v52.0.0/config/package-json.mdx:18]:
```suggestion
By specifying the library under the `install.exclude` array in the **package.json** file, you can exclude it from the version checks:
```

File: docs/pages/workflow/using-libraries.mdx [hunk 9]
```diff
@@ -123,3 +123,7 @@ If the module is not supported in [Expo Go](https://expo.dev/go), you can create
   description="Learn how to add custom native code to your Expo project."
   href="/workflow/customizing/"
 />
+
+### Excluding a third-party library from version checks
+
+If you have a specific version of a third-party library installed and want it to be excluded from version checks performed by `npx expo install` or `npx expo-doctor` or `npx expo start`, use the [`expo.install.exclude`](/versions/latest/config/package-json/#exclude) property in the **package.json** file.
```
Comment (keith-kurak) [inline docs/pages/workflow/using-libraries.mdx:129]:
```suggestion
If you have a specific version of a third-party library installed and want it to be excluded from version checks performed by `npx expo install`, `npx expo-doctor`, or `npx expo start`, use the [`expo.install.exclude`](/versions/latest/config/package-json/#exclude) property in the **package.json** file.
```

**Review by keith-kurak — APPROVED** [review]
just a few grammar-ish things!

File: docs/pages/versions/unversioned/config/package-json.mdx (hunk 3 above)
Comment (amandeepmittal) [reply docs/pages/versions/unversioned/config/package-json.mdx:3]: It's not, but let's go with it.

**Pull request merged**